### PR TITLE
Smoke e2e testing

### PR DESCRIPTION
**Pull request by narze**
# Description

- Add ability to smoke testing the bot with real discord server

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Refactor or other
  - ENVs must be added to Github Actions in order to run the smoke tests

## Screenshot

image

# Checklist:

- [x] I have run `pnpm format` and my code don't have any linting issues
  
  
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by narze** [comment]
Passing github action in my fork https://github.com/narze/kaogeek-discord-bot/actions/runs/5046874341/jobs/9052978052

**Comment by dtinth** [comment]
@narze [you are now admin](https://discord.com/channels/1107893321440964628/1108022519706943558/1110524360093077625), please feel free to add the secret by yourself krubb

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -63,6 +63,30 @@ jobs:
         if: github.event_name == 'pull_request'
         uses: davelosert/vitest-coverage-report-action@v1
 
+  smoke-e2e-check:
+    name: Run Smoke Test
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (narze) [reply .github/workflows/checks.yml]: I've test and it does not allow concurrency, so I add `concurrent: smoke-test`

**Comment by narze** [comment]
It does not let me use the added secrets, maybe github security for secrets?

image

Compare to my repo's secret
image

**Comment by dtinth** [comment]
@narze oh, this is because secrets will not run when PR comes from forked repo krub.

i think maybe smoke test has to be disabled from forked prs krub.

**Comment by narze** [comment]
Ok I'll open another PR 😭

**Pull request closed without merging**